RLBotDiscordBot/cogs/admin.py

```diff
@@ -62,6 +62,46 @@ def __init__(self, bot):
                 json.dump(self.bot.settings, f, indent=4)
 
             await ctx.send(deleted_text)
+    
+    @commands.command()
+    async def add_whitelist(self, ctx, name, *, output):
+        if not self.check_perms(ctx):
+            return
+        
+        exists = name in self.bot.settings['whitelists']
+        if exists:
+            await ctx.send("Link already whitelisted")
+
+        self.bot.settings['whitelists'][name] = output
+        with open('./RLBotDiscordBot/settings.json', 'w') as file:
+            json.dump(self.bot.settings, file, indent=4)
+
+        settings = open('./RLBotDiscordBot/settings.json', 'r')
+        self.bot.settings = json.load(settings)
```

Comment:
Why are you dumping to disk and then immediately rereading? Shouldn't `self.bot.settings` be identical to what you just dumped since you dumped it one line ago?

Reply:
You also never close the file handle.